module-services/service-db/agents/quotes/QuotesSettingsSerializer.cpp

```diff
@@ -11,17 +11,28 @@ namespace Quotes
     auto QuotesSettingsSerializer::serialize(const IdList &list) -> std::string
     {
         std::string output;
-        for (const auto &item : list) {
-            output.append(utils::to_string(item) + ',');
+        for (const auto &[type, id] : list) {
+            output.append(utils::to_string(static_cast<int>(type)) + ',');
+            output.append(utils::to_string(id) + ',');
         }
         return output;
     }
     auto QuotesSettingsSerializer::deserialize(const std::string &listString) -> IdList
     {
         std::stringstream ss(listString);
         IdList list;
+        QuoteID quoteID;
+        bool type = true;
         for (int i; ss >> i;) {
-            list.push_back(i);
+            if (type) {
+                quoteID.first = static_cast<QuoteType>(i);
+                type          = false;
```

Comment:
please use the `enum_cast` from the `magic_enum` - it handles the incompatibility.

Reply:
The `enum_cast` converts string to enum. Here I specifically wanted to keep it as int to provide minimal footprint as the whole serialized randomized order could be really long.

Reply:
Also it is easier to extract it from stringstream as all values are ints.

Reply:
right - my fault. But what if (by mistake) the value till be different than the enum scope? :D

Reply:
Checked and it will cast it to int out of scope of enum which will be simply not processed afterwards.